libvast/src/system/sink_command.cpp

```diff
@@ -76,6 +78,22 @@ caf::message sink_command(const command& cmd, actor_system& sys, caf::actor snk,
     self->send_exit(snk, exit_reason::user_shutdown);
     return caf::make_message(std::move(err));
   }
+  self->request(node, infinite, get_atom::value)
+    .receive(
+      [&](const std::string& id, system::registry& reg) {
+        // Assign accountant to sink.
+        VAST_DEBUG(&cmd, "assigns accountant from node", id, "to new sink");
+        auto er = reg.components[id].find("accountant");
+        if (er != reg.components[id].end()) {
+          auto accountant = er->second.actor;
+          self->send(snk, actor_cast<accountant_type>(accountant));
+        }
+      },
+      [&](error& e) { err = std::move(e); });
+  if (err) {
+    self->send_exit(snk, exit_reason::user_shutdown);
+    return caf::make_message(std::move(err));
+  }
```

Comment:
Is this correct: if we export with `-n`, a sink will announce an accountant twice, right? Once by looking in the registry and once through this command.

If that's the case, then we may have to perform this request only when we're in client-server mode.

(I've just looked: the same would apply to the source command, by the way.)

Reply:
You are correct, I'll remove the direct registry access in sink (and source). I don't think we should keep the extra code path if this one works for every case.